packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -0,0 +1,230 @@
+import { Template } from '../../../assertions';
+import * as logs from '../../../aws-logs';
+import * as s3 from '../../../aws-s3';
+import * as s3deploy from '../../../aws-s3-deployment';
+import * as cdk from '../../../core';
+import { CustomResourceConfig } from '../../lib/custom-resource-config/custom-resource-config';
+
+describe('when a singleton-backed custom resource does not have logging defined', () => {
+  test('addLogRetentionLifetime creates a new log group with the correct retention period if one does not already exist', () => {
```

Comment:
Organize these into:

```
// GIVEN
This is the setup for the test. For this test, this is the creation of the app, stack, the bucket, the `BucketDeployment`, and the `customResourceLogRetention` variable.

// WHEN
This is where the behavior or event you're interested in testing executes. In this case, it's a single event, `CustomResourceConfig.of()....`

// THEN
This is where your assertions are. So in this case, its the `Template.fromStack()` and everything that happens on `template`. 
```

This test file has some good examples of this: https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk/test/cdk-toolkit.test.ts#L256